Shared/Extensions/UI Extensions/UICollectionView+Extensions.swift

```diff
@@ -10,8 +10,17 @@ import Foundation
 
 extension UICollectionView {
 
-    /// Returns the indexpath of the item whose cell is closest to the center of this collectionview. If there are no items visible, then nil is returned.
+    /// Returns the indexpath of the item whose cell is closest to the center of this collectionview. If there are no cells visible, then nil is returned.
     func getCentermostVisibleIndex() -> IndexPath? {
+        if let closestCell = self.getCentermostVisibleCell() {
+            return self.indexPath(for: closestCell)
+        }
+
+        return nil
+    }
+
+    /// Gets the index of the visible cell that is closest to the center of this collection view.
+    func getCentermostVisibleCell() -> UICollectionViewCell? {
```

Comment:
Adding a convenience function to get the centermost cell, not just the index path. Note that the index path version uses this new function so we don't duplicate logic.

Reply:
Nice